### ISSUE TITLE
Compilation fails on GHC 8.10 due to unary tuples.

### ISSUE THREAD

**Issue by galenhuntington** [nikita-volkov/contravariant-extras#6]
I tried to build this package using GHC-8.10.1-alpha1, and got this:

```
library/Contravariant/Extras/Contrazip.hs:20:1: error:
    • Occurs check: cannot construct the infinite type: a1 ~ Unit a1
      Expected type: f (Unit a1)
        Actual type: f a1
    • In the second argument of ‘Data.Functor.Contravariant.Divisible.divide’, namely
        ‘f1’
      In the expression:
        ((Data.Functor.Contravariant.Divisible.divide
            (\ (_x1_0, _x2_1) -> (Unit _x1_0, Unit _x2_1)))
           f1)
          f2
      In an equation for ‘contrazip2’:
          contrazip2 f1 f2
            = ((Data.Functor.Contravariant.Divisible.divide
                  (\ (_x1_0, _x2_1) -> (Unit _x1_0, Unit _x2_1)))
                 f1)
                f2
    • Relevant bindings include
        f1 :: f a1
          (bound at library/Contravariant/Extras/Contrazip.hs:20:1)
        contrazip2 :: f a1 -> f a2 -> f (a1, a2)
          (bound at library/Contravariant/Extras/Contrazip.hs:20:1)
   |
20 | return (join (map (TH.divisibleContrazipDecs "contrazip") (reverse [2..2])))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I assume the problem is due to the change in recognition of unary tuples, probably traceable to this patch: https://gitlab.haskell.org/ghc/ghc/merge_requests/2005

Particularly, `TupleTH.splitTupleAt _ 1` emits 1-tuples when it formerly produced plain values.

I made a perfunctory effort to fix this myself, but didn't get very far (probably since I'm no good at TH).

Because of this issue, dependent packages, particularly `rebase`, cannot be built.

**Comment by nikita-volkov** [comment]
Thanks. I'll address this in the coming days

**Comment by nikita-volkov** [comment]
I'm failing to build the compiler using Stack.

I've tried putting the following in `stack.yaml`:

```yaml
resolver: ghc-git-1c071c54c712659f76f47eaddc1dd390822752bd-quickest
```

And I get the following error:

```
Error when running Shake build system:
  at want, called at src/Main.hs:89:30 in main:Main
* Depends on: binary-dist
  at apply1, called at src/Development/Shake/Internal/Rules/Rerun.hs:41:5 in shake-0.18.3-K6N2eLzVtai6jtZeWAmEIo:Development.Shake.Internal.Rules.Rerun
* Depends on: binary-dist-dir
  at need, called at src/Rules/BinaryDist.hs:124:9 in main:Rules.BinaryDist
* Depends on: docs
  at need, called at src/Rules/Documentation.hs:112:9 in main:Rules.Documentation
* Depends on: _build/docs/pdfs/Haddock.pdf
  at error, called at src/Builder.hs:335:43 in main:Builder
* Raised the exception:
Non optional builder 'xelatex' is not specified in 'hadrian/cfg/system.config' file.
```

We'll have to wait for Stack to support this compiler for me to fix this.

**Comment by galenhuntington** [comment]
I was able to use the compiler via Stack on my machine by adding these lines:

```
compiler: ghc-8.10.0.20191210  
setup-info:  
   ghc:  
      linux64:  
         8.10.0.20191210:  
            url: "https://downloads.haskell.org/~ghc/8.10.1-alpha2/ghc-8.10.0.20191210-x86_64-deb9-linux-dwarf.tar.xz"
```

There is a general config for multiple platforms here (which I've not tried):
https://github.com/DanBurton/stack-setup-info-gen/blob/master/output/stack-ghc-8.10.1-alpha2.yaml

**Comment by nikita-volkov** [comment]
Thanks! That was helpful. Got to building

**Comment by nikita-volkov** [comment]
Should be fixed in 0.3.5.1

**Comment by galenhuntington** [comment]
Yes, it's building now.  Thanks!
